File: PRNumberBiz/app/src/main/java/com/pplus/prnumberbiz/apps/mobilegift/ui/MobileGiftDetailActivity.java
package com.pplus.prnumberbiz.apps.mobilegift.ui;

import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.NonNull;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import com.pple.pplus.utils.part.format.FormatUtil;
import com.pple.pplus.utils.part.utils.StringUtils;
import com.pplus.prnumberbiz.Const;
import com.pplus.prnumberbiz.R;
import com.pplus.prnumberbiz.apps.common.toolbar.ImplToolbar;
import com.pplus.prnumberbiz.apps.common.toolbar.OnToolbarListener;
import com.pplus.prnumberbiz.apps.common.toolbar.ToolbarOption;
import com.pplus.prnumberbiz.apps.common.ui.base.BaseActivity;
import com.pplus.prnumberbiz.core.network.model.dto.MobileGift;
import com.pplus.prnumberbiz.core.network.model.dto.MobileGiftPurchase;
import com.pplus.prnumberbiz.core.util.PplusCommonUtil;

public class MobileGiftDetailActivity extends BaseActivity implements ImplToolbar, View.OnClickListener{

    @Override
    public String getPID(){

        return null;
    }

    @Override
    public int getLayoutRes(){

        return R.layout.activity_mobile_gift_detail;
    }

    private MobileGift mMobileGift;
    private int mCount;
    private MobileGiftPurchase mPurchase;
    private TextView text_count;

    @Override
    public void initializeView(Bundle savedInstanceState){

        mMobileGift = getIntent().getParcelableExtra(Const.DATA);
        ImageView image = (ImageView) findViewById(R.id.image_mobile_gift_detail);
        TextView text_name = (TextView) findViewById(R.id.text_mobile_gift_detail_name);
        TextView text_price = (TextView) findViewById(R.id.text_mobile_gift_detail_price);
        TextView text_use_area = (TextView) findViewById(R.id.text_mobile_gift_use_area);
        text_count = (TextView) findViewById(R.id.texT_mobile_gift_detail_count);
        findViewById(R.id.image_mobile_gift_detail_minus).setOnClickListener(this);
        findViewById(R.id.image_mobile_gift_detail_plus).setOnClickListener(this);
        TextView text_use_note = (TextView) findViewById(R.id.text_mobile_gift_detail_use_note);
        TextView text_use_limit = (TextView) findViewById(R.id.text_mobile_gift_detail_use_limit);
        findViewById(R.id.text_mobile_gift_detail_use_guide).setOnClickListener(this);
        findViewById(R.id.text_mobile_gift_detail_purchase).setOnClickListener(this);

        Glide.with(this).load(mMobileGift.getViewImage1()).apply(new RequestOptions().centerCrop().placeholder(R.drawable.prnumber_default_img).error(R.drawable.prnumber_default_img)).into(image);
        text_name.setText(mMobileGift.getName());
        text_price.setText(PplusCommonUtil.Companion.fromHtml(getString(R.string.html_money_unit, FormatUtil.getMoneyType(String.valueOf(mMobileGift.getSalesPrice())))));
        text_use_area.setText(mMobileGift.getUseArea());
        mCount = 1;
        setCount();

        if(StringUtils.isEmpty(mMobileGift.getUseNote())) {
            findViewById(R.id.layout_mobile_gift_detail_use_note).setVisibility(View.GONE);
        } else {
            findViewById(R.id.layout_mobile_gift_detail_use_note).setVisibility(View.VISIBLE);
            text_use_note.setText(mMobileGift.getUseNote());
        }

        if(StringUtils.isEmpty(mMobileGift.getUseLimit())) {
            findViewById(R.id.layout_mobile_gift_detail_use_limit).setVisibility(View.GONE);
        } else {
            findViewById(R.id.layout_mobile_gift_detail_use_limit).setVisibility(View.VISIBLE);
            text_use_limit.setText(mMobileGift.getUseLimit());
        }

        mPurchase = new MobileGiftPurchase();
        mPurchase.setMobileGift(mMobileGift);
    }

    private void setCount(){

        text_count.setText("" + mCount);
    }

    @Override
    public void onClick(View view){

        switch (view.getId()) {
            case R.id.image_mobile_gift_detail_minus:
                if(mCount > 1) {
                    mCount--;
                    setCount();
                }
                break;
            case R.id.image_mobile_gift_detail_plus:
                mCount++;
                setCount();
                break;
            case R.id.text_mobile_gift_detail_use_guide:
                Intent intent = new Intent(this, MobileGiftGuideActivity.class);
                intent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_CLEAR_TOP);
                startActivity(intent);
                break;
            case R.id.text_mobile_gift_detail_purchase:
                mPurchase.setCountPerTarget(mCount);
                intent = new Intent(this, MobileGiftPurchaseActivity.class);
                intent.putExtra(Const.DATA, mPurchase);
                intent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_CLEAR_TOP);
                startActivity(intent);
                break;
        }
    }

    @NonNull
    @Override
    public ToolbarOption getToolbarOption(){

        ToolbarOption toolbarOption = new ToolbarOption(this);
        toolbarOption.initializeDefaultToolbar(getString(R.string.word_product_detail), ToolbarOption.ToolbarMenu.LEFT);
        return toolbarOption;
    }

    @Override
    public OnToolbarListener getOnToolbarClickListener(){

        return new OnToolbarListener(){

            @Override
            public void onClick(View v, ToolbarOption.ToolbarMenu toolbarMenu, Object tag){

                switch (toolbarMenu) {
                    case LEFT:
                        if(tag.equals(1)) {
                            onBackPressed();
                        }
                        break;
                }
            }
        };
    }
}

File: PRNumber/app/src/main/java/com/pplus/prnumberuser/apps/mobilegift/ui/MobileGiftDetailActivity.kt
//package com.pplus.prnumberuser.apps.mobilegift.ui
//
//import android.animation.Animator
//import android.animation.AnimatorListenerAdapter
//import android.animation.ObjectAnimator
//import android.app.Activity
//import android.content.Intent
//import android.os.Bundle
//import android.view.View
//import android.view.animation.AccelerateDecelerateInterpolator
//import com.bumptech.glide.Glide
//import com.bumptech.glide.request.RequestOptions
//import com.pplus.utils.part.format.FormatUtil
//import com.pplus.prnumberuser.Const
//import com.pplus.prnumberuser.R
//import com.pplus.prnumberuser.apps.common.mgmt.LoginInfoManager
//import com.pplus.prnumberuser.apps.common.toolbar.ImplToolbar
//import com.pplus.prnumberuser.apps.common.toolbar.OnToolbarListener
//import com.pplus.prnumberuser.apps.common.toolbar.ToolbarOption
//import com.pplus.prnumberuser.apps.common.toolbar.ToolbarOption.ToolbarMenu
//import com.pplus.prnumberuser.apps.common.ui.base.BaseActivity
//import com.pplus.prnumberuser.core.network.model.dto.MobileGift
//import com.pplus.prnumberuser.core.network.model.dto.MobileGiftPurchase
//import com.pplus.prnumberuser.core.network.model.dto.MsgTarget
//import com.pplus.prnumberuser.core.util.PplusCommonUtil
//import kotlinx.android.synthetic.main.activity_mobile_gift_detail.*
//
//class MobileGiftDetailActivity : BaseActivity(), ImplToolbar {
//    override fun getPID(): String {
//        return "Main_bolpoint_pointshop_detail"
//    }
//
//    override fun getLayoutRes(): Int {
//        return R.layout.activity_mobile_gift_detail
//    }
//
//    private var mMobileGift: MobileGift? = null
//    private var mCount = 0
//    private var mPurchase: MobileGiftPurchase? = null
//    override fun initializeView(savedInstanceState: Bundle?) {
//        mMobileGift = intent.getParcelableExtra(Const.DATA)
//
//        Glide.with(this).load(mMobileGift!!.baseImage).apply(RequestOptions().centerCrop().placeholder(R.drawable.prnumber_default_img).error(R.drawable.prnumber_default_img)).into(image_mobile_gift_detail)
//        text_mobile_gift_detail_name.text = mMobileGift!!.name
//        text_mobile_gift_detail_price.text = PplusCommonUtil.fromHtml(getString(R.string.html_cash_unit2, FormatUtil.getMoneyType(mMobileGift!!.salesPrice.toString())))
//
//        text_mobile_gift_detail_use_note.text = mMobileGift!!.useNote
//
//        image_mobile_gift_detail_option_minus.setOnClickListener {
//            if (mCount > 1) {
//                mCount--
//                setCount()
//            }
//
//        }
//
//        image_mobile_gift_detail_option_plus.setOnClickListener {
//            mCount++
//            setCount()
//        }
//
//        image_mobile_gift_detail_option_close.setOnClickListener {
//            scrollOutAnimation()
//        }
//
//        text_mobile_gift_detail_buy.setOnClickListener {
////            text_mobile_gift_detail_buy.visibility = View.VISIBLE
////            text_mobile_gift_detail_option_gift.visibility = View.GONE
//            scrollInAnimation()
//        }
//
//        text_mobile_gift_detail_gift.setOnClickListener {
////            text_mobile_gift_detail_buy.visibility = View.GONE
////            text_mobile_gift_detail_option_gift.visibility = View.VISIBLE
//            scrollInAnimation()
//        }
//
//        text_mobile_gift_detail_option_gift.setOnClickListener {
//            if (!PplusCommonUtil.loginCheck(this)) {
//                return@setOnClickListener
//            }
//            val intent = Intent(this, SelectContactActivity::class.java)
//            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
//            startActivityForResult(intent, Const.REQ_SEARCH)
//        }
//
//        text_mobile_gift_detail_option_buy.setOnClickListener {
//            if (!PplusCommonUtil.loginCheck(this)) {
//                return@setOnClickListener
//            }
//            mPurchase!!.countPerTarget = mCount
//            val targetList = arrayListOf<MsgTarget>()
//            val target = MsgTarget()
//            target.mobile = LoginInfoManager.getInstance().user.mobile
//            target.name = LoginInfoManager.getInstance().user.nickname
//            targetList.add(target)
//            mPurchase!!.targetList = targetList
//
//            val totalPrice = mCount * mMobileGift!!.salesPrice
//            mPurchase!!.totalCost = totalPrice
//            mPurchase!!.pgCost = 0L
//
//            val intent = Intent(this, MobileGiftBuyActivity::class.java)
//            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
//            intent.putExtra(Const.DATA, mPurchase)
//            intent.putExtra(Const.IS_ME, true)
//            startActivityForResult(intent, Const.REQ_BUY)
//        }
//
//
//
//        mCount = 1
//        setCount()
//        mPurchase = MobileGiftPurchase()
//        mPurchase!!.mobileGift = mMobileGift
//    }
//
//    private fun setCount() {
//        text_mobile_gift_detail_option_count.text = mCount.toString()
//        val totalPrice = mCount * mMobileGift!!.salesPrice
//        text_mobile_gift_detail_option_price.text = PplusCommonUtil.fromHtml(getString(R.string.html_total_price2, FormatUtil.getMoneyType(totalPrice.toString())))
//    }
//
//    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
//        super.onActivityResult(requestCode, resultCode, data)
//        when (requestCode) {
//            Const.REQ_SEARCH -> {
//                if (resultCode == Activity.RESULT_OK) {
//                    if (data != null) {
//                        val target = data.getParcelableExtra<MsgTarget>(Const.DATA)
//                        mPurchase!!.countPerTarget = mCount
//                        val targetList = arrayListOf<MsgTarget>()
//                        targetList.add(target)
//                        mPurchase!!.targetList = targetList
//
//                        val totalPrice = mCount * mMobileGift!!.salesPrice
//                        mPurchase!!.totalCost = totalPrice
//                        mPurchase!!.pgCost = 0L
//
//                        val intent = Intent(this, MobileGiftBuyActivity::class.java)
//                        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
//                        intent.putExtra(Const.DATA, mPurchase)
//                        intent.putExtra(Const.IS_ME, false)
//                        startActivityForResult(intent, Const.REQ_BUY)
//                    }
//                }
//            }
//        }
//    }
//
//    var animating = false
//
//    private fun scrollOutAnimation() {
//        val scrollOutAnimator = ObjectAnimator.ofFloat(layout_mobile_gift_detail_option, "translationY", 0f,
//                layout_mobile_gift_detail_option.height.toFloat()).apply {
//            duration = 300
//            interpolator = AccelerateDecelerateInterpolator()
//            addListener(object : AnimatorListenerAdapter() {
//                override fun onAnimationStart(animation: Animator?) {
//                    animating = true
//                }
//
//                override fun onAnimationCancel(animation: Animator?) {
//                    animating = true
//                }
//
//                override fun onAnimationEnd(animation: Animator?) {
//                    animating = false
//                    layout_mobile_gift_detail_option.visibility = View.GONE
//                }
//            })
//        }
//
//        if (!animating) {
//            scrollOutAnimator.start()
//        }
//    }
//
//    private fun scrollInAnimation() {
//        val scrollInAnimator = ObjectAnimator.ofFloat(layout_mobile_gift_detail_option, "translationY",
//                resources.getDimension(R.dimen.height_840), 0f).apply {
//            duration = 300
//            interpolator = AccelerateDecelerateInterpolator()
//            addListener(object : AnimatorListenerAdapter() {
//                override fun onAnimationStart(animation: Animator?) {
//
//                    layout_mobile_gift_detail_option.visibility = View.VISIBLE
//                    animating = true
//                }
//
//                override fun onAnimationCancel(animation: Animator?) {
//                    animating = true
//                }
//
//                override fun onAnimationEnd(animation: Animator?) {
//                    animating = false
//                }
//            })
//        }
//
//        if (!animating) {
//            scrollInAnimator.start()
//        }
//    }
//
//    override fun getToolbarOption(): ToolbarOption {
//        val toolbarOption = ToolbarOption(this)
//        toolbarOption.initializeDefaultToolbar(getString(R.string.word_product_detail), ToolbarMenu.LEFT)
//        return toolbarOption
//    }
//
//    override fun getOnToolbarClickListener(): OnToolbarListener {
//        return OnToolbarListener { v, toolbarMenu, tag ->
//            when (toolbarMenu) {
//                ToolbarMenu.LEFT -> if (tag == 1) {
//                    onBackPressed()
//                }
//            }
//        }
//    }
//}